### PR TITLE
remove systemd-python from pypi requirements and list in bindep

### PR DESCRIPTION
**Pull request by thedoubl3j**
systemd-python is not available on pip, this causes issues when a user attempts to build an EE from the collection. the package is available via dnf so moving the dependency to a bindep.txt will pass the dep to dnf during the ansible-builder build phase which will be able to resolve the dep. see #153 for more context on the issue

### CODE REVIEW

**Review by benthomasson — APPROVED** [review]

**Review by bzwei — APPROVED** [review]

**Review by Alex-Izquierdo — CHANGES_REQUESTED** [review]
I don't think to say that systemd-package is unavailable on pypi or wrong packaged is correct. As is described in the docs, it just  needs to have installed several system dependencies. 

As this process is complicated, I agree with remove it from the requirements.txt. But the readme must indicate that `python3-systemd` package must be installed in addition to the requirements.txt.

File: bindep.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+systemd-python
```
Comment (Alex-Izquierdo) [inline bindep.txt:1]: According [to the documentation](https://github.com/systemd/python-systemd#installation) and my own tests, the package name is `python3-systemd`

**Review by Alex-Izquierdo — CHANGES_REQUESTED** [review]

**Comment by Alex-Izquierdo** [comment]
Hi @thedoubl3j thanks for your contribution and update the right package name. Readme needs to be modified too, to indicate that `python3-systemd` package must be installed, otherwise the users that follow these instructions would have unexpected errors.

**Comment by thedoubl3j** [comment]
@Alex-Izquierdo done and squashed commits as well

File: README.md [hunk 2]
```diff
@@ -23,6 +23,12 @@ The python module dependencies are not installed by ansible-galaxy. They must be
 pip install -r requirements.txt
 ```
 
+The binary module dependencies are not installed by ansible-galaxy. They must also  be installed manually using dnf:
```
Comment (konstruktoid) [inline README.md:26]: Write a generic instruction or add additional instructions using other package managers as well?

File: README.md [hunk 3]
```diff
@@ -23,6 +23,12 @@ The python module dependencies are not installed by ansible-galaxy. They must be
...
+
+```
+dnf install python3-systemd
+```
+
```
Comment (Alex-Izquierdo) [inline README.md:31]: IMO this test is clearer:
```suggestion
To use the journald source plugin, it is also necessary to have the system package named `python3-systemd` installed on Debian or Fedora operating systems. For other platforms, loot at the [systemd-python documentation](https://github.com/systemd/python-systemd#installation). 
```

Comment (Alex-Izquierdo) [reply README.md:31]: Hi @thedoubl3j Do you agree to commit my suggestion?

**Comment by Alex-Izquierdo** [comment]
package python3-systemd must be installed as well in the CI workflows to pass the tests.

**Comment by ssbarnea** [comment]
While use of `bindep.txt` seems nice (i wanted to add it myself), I am closing this because referred bug was incorrect. Note that I am in favour of adding a bindep.txt fine, but if we do, it must be accompanied by some GHA/tox changes that would be testing it for supported platforms.

**Pull request closed without merging**